vividus-extension-selenium/src/main/java/org/vividus/selenium/AbstractVividusWebDriverFactory.java

```diff
@@ -80,12 +85,34 @@ private void setDesiredCapabilities(DesiredCapabilities desiredCapabilities)
                                         .map(scenario -> scenario.getMeta().inheritFrom(storyMeta))
                                         .orElse(storyMeta);
             MetaWrapper metaWrapper = new MetaWrapper(mergedMeta);
-            ControllingMetaTag.setDesiredCapabilitiesFromMeta(desiredCapabilities, metaWrapper);
+            ControllingMetaTag.setDesiredCapabilitiesFromMeta(mergedCapabilities, metaWrapper);
         }
+
+        return mergedCapabilities;
     }
 
     protected abstract WebDriver createWebDriver(DesiredCapabilities desiredCapabilities);
 
+    private static DesiredCapabilities mergeCapabilities(DesiredCapabilities baseCapabilities,
+            DesiredCapabilities sourceCapabilities)
+    {
+        Map<String, Object> mergedCapabilities = merge(baseCapabilities.asMap(), sourceCapabilities.asMap());
+        return new DesiredCapabilities(mergedCapabilities);
+    }
+
+    @SuppressWarnings({ "unchecked", "rawtypes" })
+    private static Map<String, Object> merge(Map<String, Object> base, Map<String, Object> source)
```

Comment:
I think we need to move some common logic merging capabilities into a separate class:
https://github.com/vividus-framework/vividus/blob/6add6da91c8abfaac9ebcb9a2738fd718915d36c/vividus-extension-selenium/src/main/java/org/vividus/selenium/AbstractWebDriverFactory.java#L139-L155